sdk/core/azure-core/src/main/java/com/azure/core/util/HttpClientOptions.java

```diff
@@ -299,21 +350,34 @@ public HttpClientOptions setConnectionIdleTimeout(Duration connectionIdleTimeout
      * @return The connection idle timeout duration.
      */
     public Duration getConnectionIdleTimeout() {
-        return getTimeout(connectionIdleTimeout);
+        return getTimeout(connectionIdleTimeout, DEFAULT_CONNECTION_IDLE_TIMEOUT);
+    }
+
+    private static Duration getDefaultTimeout(String property, Duration defaultTimeout) {
+        String envTimeout = Configuration.getGlobalConfiguration().get(property);
+        if (CoreUtils.isNullOrEmpty(envTimeout)) {
+            return defaultTimeout;
+        }
+
+        try {
+            return Duration.ofMillis(Integer.parseInt(envTimeout));
+        } catch (NumberFormatException ex) {
+            return defaultTimeout;
+        }
```

Comment:
This can be made public and used in OkHttpClientBuilder and NettyClientBuilder.

Reply:
Where would you expect this to be declared? `CoreUtils`?

Reply:
Yeah, this can be moved to CoreUtils.